shared/src/main/scala/scodec/codecs/VariableSizeDelimitedCodec.scala

```diff
@@ -0,0 +1,46 @@
+package scodec
+package codecs
+
+import scodec.bits.BitVector
+
+private[codecs] final class VariableSizeDelimitedCodec[A](delimiterCodec: Codec[Unit], valueCodec: Codec[A], multipleValueSize: Long = 0L) extends Codec[A] {
+
+  val delimiter = delimiterCodec.encode(()).require
+  val segmentSize = valueCodec.sizeBound.exact.getOrElse(multipleValueSize)
+
+  require(segmentSize > 0, "valueCodec must have an exact sizeBound or you need to specify multipleValueSize")
+
+  require(delimiterCodec.sizeBound.lowerBound >= segmentSize, "delimiterCodec cannot be smaller than the sizeBound of the valueCodec")
+
+  def sizeBound = delimiterCodec.sizeBound.atLeast
+
+  override def encode(a: A) = for {
+    encA <- valueCodec.encode(a)
+    encDel <- delimiterCodec.encode(())
+  } yield encA ++ encDel
+
+  override def decode(buffer: BitVector) = {
+    val index = findDelimiterIndex(buffer) 
+    if (index != -1) {
+      val valueBuffer = buffer.take(index)
+      val remainder = buffer.drop(index + delimiter.size)
+      println(s"VariableSizeDelimitedCodec.decode: [${delimiter.size}] $index [${valueBuffer.size}]$valueBuffer [${remainder.size}]$remainder")
```

Comment:
Let's remove this line and add any necessary information to the failure message.